### ISSUE TITLE
Missing PriorityClass

### ISSUE THREAD

**Issue by jpkrohling** [openshift/cluster-logging-operator#41]
I'm trying to get a simple CR deployed, but I'm facing the following error:

```
ERRO[0031] error syncing key (openshift-logging/example): Unable to create or update collection: Failure creating Collection priority class: failed to get resource client: failed to get resource type: failed to get the resource REST mapping for GroupVersionKind(scheduling.k8s.io/v1beta1, Kind=PriorityClass): no matches for kind "PriorityClass" in version "scheduling.k8s.io/v1beta1" 
```

Here are the steps I'm following, based on the readme and trial and error.
```
$ minishift version
v1.25.0+90fb23e
$ minishift start --cpus 2 --memory 8192
$ oc login -u system:admin
$ oc create -f manifests/01-namespace.yaml
$ oc project openshift-logging
$ make deploy-setup
$ REPO_PREFIX=openshift/ \
    IMAGE_PREFIX=origin- \
    OPERATOR_NAME=cluster-logging-operator \
    WATCH_NAMESPACE=openshift-logging \
    KUBERNETES_CONFIG=~/.kube/config \
    ELASTICSEARCH_IMAGE=docker.io/openshift/origin-logging-elasticsearch5:latest \
    OAUTH_PROXY_IMAGE=docker.io/openshift/oauth-proxy:latest \
    KIBANA_IMAGE=docker.io/openshift/origin-logging-kibana5:latest \
    CURATOR_IMAGE=docker.io/openshift/origin-logging-curator5:latest \
    FLUENTD_IMAGE=docker.io/openshift/origin-logging-fluentd:latest \
    go run cmd/cluster-logging-operator/main.go
$ oc apply -f hack/cr.yaml
```

What am I doing wrong?

**Comment by josefkarasek** [comment]
I wonder if PriorityClass is released already..
Maybe you need to enable experimental features for minishift 
https://docs.okd.io/latest/minishift/using/experimental-features.html#enabling-experimental-oc-flags

https://github.com/abhinavsinha001/website/blob/446b9cb33556abb663c02554153e80dfc531d79a/docs/concepts/configuration/pod-priority-preemption.md#enabling-priority-and-preemption

**Comment by jpkrohling** [comment]
Do you have an example of a working cluster? I can't seem to be able to transform the two documents into valid commands :-/ 

I tried the following and it didn't work:

```
MINISHIFT_ENABLE_EXPERIMENTAL=y minishift-start --extra-clusterup-flags "--enable=*,service-catalog --feature-gates=PodPriority=true --runtime-config=scheduling.k8s.io/v1alpha1=true --admission-control=Priority"
```

It fails with:

```
command : /var/lib/minishift/bin/oc cluster up --image 'openshift/origin-${component}:v3.10.0' --public-hostname 192.168.42.190 --routing-suffix 192.168.42.190.nip.io --base-dir /var/lib/minishift/base --enable=*,service-catalog --feature-gates=PodPriority=true --runtime-config=scheduling.k8s.io/v1alpha1=true --admission-control=Priority
err     : exit status 1
output  : Error: unknown flag: --feature-gates
```

Removing the unknown flag yields a failure on another unknown flag:

```
output  : Error: unknown flag: --runtime-config
```

And then:
```
output  : Error: unknown flag: --admission-control
```

Removing that will, well, not install the Priority feature :)

**Comment by josefkarasek** [comment]
I personally use ec2 instance where I can just edit master-config.yaml and restart systemd svc / system container.
Do you have some other way how to deploy origin? Eg local libvirt machine?

**Comment by jpkrohling** [comment]
`minishift` does provision a virtual machine with libvirt:

```
-- Starting profile 'istio-tutorial'
-- Check if deprecated options are used ... OK
-- Checking if https://github.com is reachable ... OK
-- Checking if requested OpenShift version 'v3.10.0' is valid ... OK
-- Checking if requested OpenShift version 'v3.10.0' is supported ... OK
-- Checking if requested hypervisor 'kvm' is supported on this platform ... OK
-- Checking if KVM driver is installed ... 
   Driver is available at /home/jpkroehling/bin/docker-machine-driver-kvm ... 
   Checking driver binary is executable ... OK
-- Checking if Libvirt is installed ... OK
-- Checking if Libvirt default network is present ... OK
-- Checking if Libvirt default network is active ... OK
-- Checking the ISO URL ... OK
-- Checking if provided oc flags are supported ... OK
-- Starting the OpenShift cluster using 'kvm' hypervisor ...
-- Minishift VM will be configured with ...
   Memory:    8 GB
   vCPUs :    2
   Disk size: 20 GB
-- Starting Minishift VM ................. OK
-- Checking for IP address ... OK
-- Checking for nameservers ... OK
-- Checking if external host is reachable from the Minishift VM ... 
   Pinging 8.8.8.8 ... OK
-- Checking HTTP connectivity from the VM ... 
   Retrieving http://minishift.io/index.html ... OK
-- Checking if persistent storage volume is mounted ... OK
-- Checking available disk space ... 1% used OK
   Importing 'openshift/origin-control-plane:v3.10.0' .............. OK
   Importing 'openshift/origin-docker-registry:v3.10.0' ........ OK
   Importing 'openshift/origin-haproxy-router:v3.10.0' ............. OK
-- OpenShift cluster will be configured with ...
   Version: v3.10.0
-- Copying oc binary from the OpenShift container image to VM ... OK
-- Starting OpenShift cluster -- Extra 'oc' cluster up flags (experimental) ... 
   '--enable=*,service-catalog'
.....................................................
Getting a Docker client ...
Checking if image openshift/origin-control-plane:v3.10.0 is available ...
Pulling image openshift/origin-cli:v3.10.0
Pulled 1/5 layers, 22% complete
Pulled 2/5 layers, 41% complete
Pulled 3/5 layers, 62% complete
Pulled 4/5 layers, 83% complete
Pulled 5/5 layers, 100% complete
Extracting
Image pull complete
Pulling image openshift/origin-node:v3.10.0
Pulled 2/6 layers, 35% complete
Pulled 3/6 layers, 52% complete
Pulled 4/6 layers, 78% complete
Pulled 5/6 layers, 90% complete
Pulled 6/6 layers, 100% complete
Extracting
Image pull complete
Checking type of volume mount ...
Determining server IP ...
Using public hostname IP 192.168.42.45 as the host IP
Checking if OpenShift is already running ...
Checking for supported Docker version (=>1.22) ...
Checking if insecured registry is configured properly in Docker ...
Checking if required ports are available ...
Checking if OpenShift client is configured properly ...
Checking if image openshift/origin-control-plane:v3.10.0 is available ...
Starting OpenShift using openshift/origin-control-plane:v3.10.0 ...
I1130 06:54:06.673631    2131 config.go:42] Running "create-master-config"
I1130 06:54:09.050840    2131 config.go:46] Running "create-node-config"
I1130 06:54:09.870594    2131 flags.go:30] Running "create-kubelet-flags"
I1130 06:54:10.493538    2131 run_kubelet.go:48] Running "start-kubelet"
I1130 06:54:10.742264    2131 run_self_hosted.go:172] Waiting for the kube-apiserver to be ready ...
I1130 06:55:19.748814    2131 interface.go:26] Installing "kube-proxy" ...
I1130 06:55:19.750033    2131 interface.go:26] Installing "kube-dns" ...
I1130 06:55:19.750061    2131 interface.go:26] Installing "openshift-apiserver" ...
I1130 06:55:19.750199    2131 apply_template.go:83] Installing "openshift-apiserver"
I1130 06:55:19.752700    2131 apply_template.go:83] Installing "kube-proxy"
I1130 06:55:19.753531    2131 apply_template.go:83] Installing "kube-dns"
I1130 06:55:22.272671    2131 interface.go:41] Finished installing "kube-proxy" "kube-dns" "openshift-apiserver"
I1130 06:55:58.311426    2131 run_self_hosted.go:224] openshift-apiserver available
I1130 06:55:58.311454    2131 interface.go:26] Installing "openshift-controller-manager" ...
I1130 06:55:58.311468    2131 apply_template.go:83] Installing "openshift-controller-manager"
I1130 06:56:00.698237    2131 interface.go:41] Finished installing "openshift-controller-manager"
Adding default OAuthClient redirect URIs ...
Adding centos-imagestreams ...
Adding registry ...
Adding service-catalog ...
Adding web-console ...
Adding router ...
Adding sample-templates ...
Adding persistent-volumes ...
I1130 06:56:00.726271    2131 interface.go:26] Installing "centos-imagestreams" ...
I1130 06:56:00.726281    2131 interface.go:26] Installing "openshift-image-registry" ...
I1130 06:56:00.726288    2131 interface.go:26] Installing "openshift-service-catalog" ...
I1130 06:56:00.726294    2131 interface.go:26] Installing "openshift-web-console-operator" ...
I1130 06:56:00.726300    2131 interface.go:26] Installing "openshift-router" ...
I1130 06:56:00.726306    2131 interface.go:26] Installing "sample-templates" ...
I1130 06:56:00.726311    2131 interface.go:26] Installing "persistent-volumes" ...
I1130 06:56:00.726731    2131 apply_list.go:68] Installing "centos-imagestreams"
I1130 06:56:00.728014    2131 apply_template.go:83] Installing "openshift-web-console-operator"
I1130 06:56:00.728398    2131 interface.go:26] Installing "sample-templates/mongodb" ...
I1130 06:56:00.728407    2131 interface.go:26] Installing "sample-templates/postgresql" ...
I1130 06:56:00.728414    2131 interface.go:26] Installing "sample-templates/django quickstart" ...
I1130 06:56:00.728421    2131 interface.go:26] Installing "sample-templates/rails quickstart" ...
I1130 06:56:00.728427    2131 interface.go:26] Installing "sample-templates/jenkins pipeline ephemeral" ...
I1130 06:56:00.728434    2131 interface.go:26] Installing "sample-templates/mariadb" ...
I1130 06:56:00.728440    2131 interface.go:26] Installing "sample-templates/mysql" ...
I1130 06:56:00.728446    2131 interface.go:26] Installing "sample-templates/cakephp quickstart" ...
I1130 06:56:00.728453    2131 interface.go:26] Installing "sample-templates/dancer quickstart" ...
I1130 06:56:00.728460    2131 interface.go:26] Installing "sample-templates/nodejs quickstart" ...
I1130 06:56:00.728467    2131 interface.go:26] Installing "sample-templates/sample pipeline" ...
I1130 06:56:00.728513    2131 apply_list.go:68] Installing "sample-templates/sample pipeline"
I1130 06:56:00.728711    2131 apply_list.go:68] Installing "sample-templates/mongodb"
I1130 06:56:00.728835    2131 apply_list.go:68] Installing "sample-templates/postgresql"
I1130 06:56:00.728952    2131 apply_list.go:68] Installing "sample-templates/django quickstart"
I1130 06:56:00.729051    2131 apply_list.go:68] Installing "sample-templates/rails quickstart"
I1130 06:56:00.729154    2131 apply_list.go:68] Installing "sample-templates/jenkins pipeline ephemeral"
I1130 06:56:00.729260    2131 apply_list.go:68] Installing "sample-templates/mariadb"
I1130 06:56:00.729360    2131 apply_list.go:68] Installing "sample-templates/mysql"
I1130 06:56:00.729454    2131 apply_list.go:68] Installing "sample-templates/cakephp quickstart"
I1130 06:56:00.729569    2131 apply_list.go:68] Installing "sample-templates/dancer quickstart"
I1130 06:56:00.729683    2131 apply_list.go:68] Installing "sample-templates/nodejs quickstart"
I1130 06:56:00.887399    2131 apply_template.go:83] Installing "service-catalog"
I1130 06:56:11.953650    2131 interface.go:41] Finished installing "sample-templates/mongodb" "sample-templates/postgresql" "sample-templates/django quickstart" "sample-templates/rails quickstart" "sample-templates/jenkins pipeline ephemeral" "sample-templates/mariadb" "sample-templates/mysql" "sample-templates/cakephp quickstart" "sample-templates/dancer quickstart" "sample-templates/nodejs quickstart" "sample-templates/sample pipeline"
I1130 06:56:50.106285    2131 interface.go:41] Finished installing "centos-imagestreams" "openshift-image-registry" "openshift-service-catalog" "openshift-web-console-operator" "openshift-router" "sample-templates" "persistent-volumes"
Login to server ...
Creating initial project "myproject" ...
Server Information ...
OpenShift server started.

The server is accessible via web console at:
    https://192.168.42.45:8443

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin


-- Applying addon 'admin-user':..
-- Applying addon 'anyuid':.
 Add-on 'anyuid' changed the default security context constraints to allow pods to run as any user.
 Per default OpenShift runs containers using an arbitrarily assigned user ID.
 Refer to https://docs.okd.io/latest/architecture/additional_concepts/authorization.html#security-context-constraints and
 https://docs.okd.io/latest/creating_images/guidelines.html#openshift-origin-specific-guidelines for more information.
```

**Comment by richm** [comment]
there is another issue about priority class - is this related?  https://github.com/openshift/cluster-logging-operator/issues/44

**Comment by jpkrohling** [comment]
> is this related?

I think it's not directly related, but if I understood @jcantrill's comment, removing the priority class from this operator would solve this one here.

**Comment by jpkrohling** [comment]
@josefkarasek, @richm is it a target of this Operator to run in `minishift` and/or `minikube`? I realize you are probably busy with other more pressing tasks, but getting this operator to run in `minishift` (or `minikube`, ideally) would unblock me from working on the Jaeger Operator's side, consuming this one here.

**Comment by richm** [comment]
@jpkrohling it should run anywhere - that is, I'm not aware of anything which would prevent this operator from working in a minishift or minikube environment, assume it works in a "regular" openshift environment.

**Comment by jpkrohling** [comment]
That's not the case right now, but I'll keep trying. If anyone of you is able to get it working, I'd appreciate if you should share the commands used to provision a cluster (Kubernetes/OpenShift) with `minishift`/`minikube`.

**Comment by richm** [comment]
> That's not the case right now, but I'll keep trying. If anyone of you is able to get it working, I'd appreciate if you should share the commands used to provision a cluster (Kubernetes/OpenShift) with `minishift`/`minikube`.

The operator should work in any environment - if it does not, that's probably a bug.  Please open a new issue and/or provide a PR (unless you are sure that the problem you are seeing is related to Missing PriorityClass).

**Comment by jpkrohling** [comment]
I'm not sure this issue is still relevant. I'm trying from scratch, with the following commands:

```
$ minishift start --cpus 2 --memory 8192
$ make deploy-example
$ make deploy
```

And I get an `ImagePullBackOff` for the pods:

`oc get pods`
```
cluster-logging-operator-7d5fc8844-5fkqg   0/1       ImagePullBackOff   0          7m
elasticsearch-operator-7d9c99d97-97rjl     0/1       ImagePullBackOff   0          7m
```

I'm clearly missing something, having a reference to the commands that are expected to get the operator running locally would be really helpful.

**Comment by richm** [comment]
@jcantrill is this how you were developing?  And this is why you needed to build the image locally and push to the image registry running in minishift?  There is a `REMOTE_REGISTRY` env. var. which I think is used in this case?  https://github.com/openshift/cluster-logging-operator/blob/master/hack/deploy.sh#L7 - I don't know how it is supposed to work

@jpkrohling What does `oc describe pod cluster-logging-operator-7d5fc8844-5fkqg` say?  Does it say which image it cannot pull?
You should only have to do https://github.com/openshift/cluster-logging-operator#full-deploy - so `make deploy-example` only

**Comment by richm** [comment]
@jcantrill is this how you were developing?  And this is why you needed to build the image locally and push to the image registry running in minishift?  There is a `REMOTE_REGISTRY` env. var. which I think is used in this case?  https://github.com/openshift/cluster-logging-operator/blob/master/hack/deploy.sh#L7 - I don't know how it is supposed to work

@jpkrohling What does `oc describe pod cluster-logging-operator-7d5fc8844-5fkqg` say?  Does it say which image it cannot pull?
You should only have to do https://github.com/openshift/cluster-logging-operator#full-deploy - so `make deploy-example` only

**Comment by jpkrohling** [comment]
I just tried on a fresh cluster set via `minishift`, running simply `make deploy-example`, and got a similar result. Here's the pod description:

```
$ oc describe pod cluster-logging-operator-7d5fc8844-4qbdq
Name:           cluster-logging-operator-7d5fc8844-4qbdq
Namespace:      openshift-logging
Node:           localhost/192.168.122.19
Start Time:     Fri, 07 Dec 2018 10:37:33 +0100
Labels:         name=cluster-logging-operator
                pod-template-hash=381974400
Annotations:    openshift.io/scc=anyuid
Status:         Pending
IP:             172.17.0.5
Controlled By:  ReplicaSet/cluster-logging-operator-7d5fc8844
Containers:
  cluster-logging-operator:
    Container ID:  
    Image:         openshift/cluster-logging-operator:latest
    Image ID:      
    Port:          <none>
    Host Port:     <none>
    Command:
      cluster-logging-operator
    State:          Waiting
      Reason:       ErrImagePull
    Ready:          False
    Restart Count:  0
    Environment:
      WATCH_NAMESPACE:      openshift-logging (v1:metadata.namespace)
      OPERATOR_NAME:        cluster-logging-operator
      ELASTICSEARCH_IMAGE:  docker.io/openshift/origin-logging-elasticsearch5:latest
      FLUENTD_IMAGE:        docker.io/openshift/origin-logging-fluentd:latest
      KIBANA_IMAGE:         docker.io/openshift/origin-logging-kibana5:latest
      CURATOR_IMAGE:        docker.io/openshift/origin-logging-curator5:latest
      OAUTH_PROXY_IMAGE:    docker.io/openshift/oauth-proxy:latest
      RSYSLOG_IMAGE:        docker.io/viaq/rsyslog:latest
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from cluster-logging-operator-token-5h5sl (ro)
Conditions:
  Type           Status
  Initialized    True 
  Ready          False 
  PodScheduled   True 
Volumes:
  cluster-logging-operator-token-5h5sl:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  cluster-logging-operator-token-5h5sl
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     <none>
Events:
  Type     Reason     Age                From                Message
  ----     ------     ----               ----                -------
  Normal   Scheduled  54s                default-scheduler   Successfully assigned cluster-logging-operator-7d5fc8844-4qbdq to localhost
  Normal   Pulling    14s (x3 over 54s)  kubelet, localhost  pulling image "openshift/cluster-logging-operator:latest"
  Warning  Failed     12s (x3 over 52s)  kubelet, localhost  Failed to pull image "openshift/cluster-logging-operator:latest": rpc error: code = Unknown desc = repository docker.io/openshift/cluster-logging-operator not found: does not exist or no pull access
  Warning  Failed     12s (x3 over 52s)  kubelet, localhost  Error: ErrImagePull
  Normal   BackOff    0s (x3 over 51s)   kubelet, localhost  Back-off pulling image "openshift/cluster-logging-operator:latest"
  Warning  Failed     0s (x3 over 51s)   kubelet, localhost  Error: ImagePullBackOff
```

**Comment by richm** [comment]
you will need to push images from your host into the registry in the cluster running in the vm

you will first need a user with a password with cluster-admin credentials.  Not sure if minishift or minikube creates one for you automatically - for example, the new openshift-installer creates a user called `kubeadmin`

with this user do `oc login --username=thename --password=password`

then `REMOTE_REGISTRY=true make deploy-example`

**Comment by jpkrohling** [comment]
Current status: it failed at a different point this time. Here's the summary:

```
minishift start --cpus 2 --memory 8192
oc login -u system:admin
oc adm policy add-cluster-role-to-user cluster-admin developer
oc adm policy add-cluster-role-to-user admin developer # not sure this one is required
oc login -u developer # won't work with system:admin, as `oc whoami -t` won't return a token, which apparently is required by the script
REMOTE_REGISTRY=true make deploy-example
```

```
++ oc get service docker-registry -n default -o 'jsonpath={.spec.clusterIP}'
+ registry_ip=172.30.1.1
+ cat manifests/05-deployment.yaml
+ oc create -f -
hack/deploy.sh: line 10: IMAGE_TAG: unbound variable
error: no objects passed to create
make: *** [Makefile:79: deploy] Error 1
```

**Comment by richm** [comment]
yep - there were some bugs in the script.  I've created two PRs to address these and other issues: https://github.com/openshift/cluster-logging-operator/pull/52 and https://github.com/openshift/elasticsearch-operator/pull/45 - they have both now merged - please give the new code a try

**Comment by jpkrohling** [comment]
Looks far better! TL;DR: there are still some road bumps, but I got it deployed. Just need to check if I can get Jaeger to use it.

Long version:

The script did finish, but I ended up with a small bug in the `elasticsearch-operator` deployment:

```
$ oc get pods
NAME                                        READY     STATUS             RESTARTS   AGE
cluster-logging-operator-57bcff474f-gr82p   1/1       Running            0          6m
elasticsearch-operator-7c797d94fc-fjm6v     0/1       InvalidImageName   0          5m
kibana-7bb78457df-qmhzx                     2/2       Running            0          5m
$ oc describe pod elasticsearch-operator-7c797d94fc-fjm6v
Name:           elasticsearch-operator-7c797d94fc-fjm6v
Namespace:      openshift-logging
Node:           localhost/192.168.122.122
Start Time:     Mon, 10 Dec 2018 13:25:50 +0100
Labels:         name=elasticsearch-operator
                pod-template-hash=3735385097
Annotations:    openshift.io/scc=anyuid
Status:         Pending
IP:             172.17.0.6
Controlled By:  ReplicaSet/elasticsearch-operator-7c797d94fc
Containers:
  elasticsearch-operator:
    Container ID:  
    Image:         172.30.1.1:5000/openshift/elasticsearch-operator:latest:latest
    Image ID:      
    Port:          60000/TCP
    Host Port:     0/TCP
    Command:
      elasticsearch-operator
    State:          Waiting
      Reason:       InvalidImageName
    Ready:          False
    Restart Count:  0
    Environment:
      WATCH_NAMESPACE:  openshift-logging (v1:metadata.namespace)
      OPERATOR_NAME:    elasticsearch-operator
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from elasticsearch-operator-token-hfv2j (ro)
Conditions:
  Type           Status
  Initialized    True 
  Ready          False 
  PodScheduled   True 
Volumes:
  elasticsearch-operator-token-hfv2j:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  elasticsearch-operator-token-hfv2j
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     <none>
Events:
  Type     Reason         Age               From                Message
  ----     ------         ----              ----                -------
  Normal   Scheduled      6m                default-scheduler   Successfully assigned elasticsearch-operator-7c797d94fc-fjm6v to localhost
  Warning  Failed         4m (x13 over 6m)  kubelet, localhost  Error: InvalidImageName
  Warning  InspectFailed  1m (x25 over 6m)  kubelet, localhost  Failed to apply default image tag "172.30.1.1:5000/openshift/elasticsearch-operator:latest:latest": couldn't parse image reference "172.30.1.1:5000/openshift/elasticsearch-operator:latest:latest": invalid reference format
```

Issuing a `oc edit deployment elasticsearch-operator` and changing the image to remove the duplicate `:latest` got me further:

```
$ oc get pods
NAME                                                  READY     STATUS    RESTARTS   AGE
cluster-logging-operator-57bcff474f-gr82p             1/1       Running   0          9m
elasticsearch-clientdatamaster-0-1-79f85bfc94-mxq7j   0/1       Error     3          1m
elasticsearch-operator-7f78cb8456-cq9f9               1/1       Running   0          1m
kibana-7bb78457df-qmhzx                               2/2       Running   0          8m
```

The error message from the `clientdatamaster` was:

```
ERROR: [1] bootstrap checks failed
[1]: max virtual memory areas vm.max_map_count [65530] is too low, increase to at least [262144]
```

Which can be solved with:

```
$ minishift ssh
[docker@istio-tutorial ~]$ sudo sysctl -w vm.max_map_count=262144
vm.max_map_count = 262144
[docker@istio-tutorial ~]$ logout
```

After deleting the pod (which was at this stage marked as `CrashLoopBackOff`), it did come up online:

```
$ oc get pods
NAME                                                  READY     STATUS    RESTARTS   AGE
cluster-logging-operator-57bcff474f-gr82p             1/1       Running   0          13m
elasticsearch-clientdatamaster-0-1-79f85bfc94-kf8kw   1/1       Running   0          1m
elasticsearch-operator-7f78cb8456-cq9f9               1/1       Running   0          5m
kibana-7bb78457df-qmhzx                               2/2       Running   0          12m
```

**Comment by richm** [comment]
>     Image:         172.30.1.1:5000/openshift/elasticsearch-operator:latest:latest

Were you using an earlier version of the PR?  The version that finally merged should not do this.

> sudo sysctl -w vm.max_map_count=262144

We don't yet have a good way to do this in the minishift/vm style deployment.  In the official 4.0 release, this is supposed to be done by the node tuning operator.  I suppose we could do something like
```
ssh $HOST? -n "sudo sysctl -w vm.max_map_count=262144"
```
but I'm not sure what to use for `$HOST?` in this case.

**Comment by jpkrohling** [comment]
> Were you using an earlier version of the PR? The version that finally merged should not do this.

I'm using the latest master as of now, a7cd215e5a4fef4c08ea7a1f501de9fb4ff354f0.

> We don't yet have a good way to do this in the minishift/vm style deployment. In the official 4.0 release, this is supposed to be done by the node tuning operator. I suppose we could do something like
> 
>     ssh $HOST? -n "sudo sysctl -w vm.max_map_count=262144"
>
> but I'm not sure what to use for $HOST? in this case.

I just tried the following and it works:

```
minishift ssh -- sudo sysctl -w vm.max_map_count=262144
```

**Comment by richm** [comment]
> > Were you using an earlier version of the PR? The version that finally merged should not do this.
> 
> I'm using the latest master as of now, [a7cd215](https://github.com/openshift/cluster-logging-operator/commit/a7cd215e5a4fef4c08ea7a1f501de9fb4ff354f0).

And you are still getting the `image:latest:latest` tag?

> 
> > We don't yet have a good way to do this in the minishift/vm style deployment. In the official 4.0 release, this is supposed to be done by the node tuning operator. I suppose we could do something like
> > ```
> > ssh $HOST? -n "sudo sysctl -w vm.max_map_count=262144"
> > ```
> > but I'm not sure what to use for $HOST? in this case.
> 
> I just tried the following and it works:
> 
> ```
> minishift ssh -- sudo sysctl -w vm.max_map_count=262144
> ```

Ok.  Please submit a PR so that if you are using minishift, it will run this.  Bonus points if you can auto-detect if minishift should be used or not.

**Comment by jpkrohling** [comment]
> And you are still getting the `image:latest:latest` tag?

Yes.

> Ok. Please submit a PR so that if you are using minishift, it will run this. Bonus points if you can auto-detect if minishift should be used or not.

I'll record an issue instead, as I'm close to going on PTO and probably won't be able to send it by then.

**Comment by jpkrohling** [comment]
I'm closing this, as the original issue isn't relevant anymore
